constants/weather_constants.asm

```diff
@@ -0,0 +1,21 @@
+; GetCurrentWeather return values
+	const_def
+	const OW_WEATHER_NONE
+	const OW_WEATHER_RAIN
+	const OW_WEATHER_SNOW
+	const OW_WEATHER_THUNDERSTORM
+	const OW_WEATHER_SANDSTORM
+DEF NUM_OW_WEATHERS EQU const_value - 1
+
+; wWeatherFlags
+	const_def
+	const OW_WEATHER_DO_FLY_F             ; 0
+	const OW_WEATHER_DISABLED_F           ; 1
+	const OW_WEATHER_LIGHTNING_DISABLED_F ; 2
+
+DEF WEATHER_TILE_1  EQU $6d
+DEF WEATHER_TILE_2  EQU $6e
+DEF RAINDROP_TILE   EQU WEATHER_TILE_1
+DEF RAINSPLASH_TILE EQU WEATHER_TILE_2
+DEF SNOWFLAKE_TILE  EQU WEATHER_TILE_1
+DEF SANDSTORM_TILE  EQU WEATHER_TILE_1
```

Comment:
These are only used in engine/overworld/weather.asm, so should be declared at its top. (Other files do the same with specific tile IDs, like judge_machine.asm, map_name_sign.asm, slot_machine.asm, etc.)